rosidl_typesupport_c/src/type_support_dispatch.hpp

```diff
@@ -34,7 +48,9 @@ namespace rosidl_typesupport_c
 
 std::string find_library_path(const std::string & library_name);
 
+#if !defined(ROSIDL_TYPESUPPORT_C_NO_ENV)
 std::string get_env_var(const char * env_var);
+#endif
```

Comment:
It might be better to leave the API in place and just always return `""`. It would be nice to have the same API on the library regardless of the ability to get env vars under the hood. Ultimately this should probably be using the `rcutils` function for getting the environment variables.